cmd/ocsp-updater/main.go

```diff
@@ -23,20 +28,80 @@ import (
 	"github.com/letsencrypt/boulder/sa"
 )
 
-// FatalError indicates the updater should stop execution
-type FatalError string
-
-func (e FatalError) Error() string { return string(e) }
-
 // OCSPUpdater contains the useful objects for the Updater
 type OCSPUpdater struct {
 	stats statsd.Statter
 	log   *blog.AuditLogger
-	cac   rpc.CertificateAuthorityClient
+
+	rpcMu *sync.RWMutex
+
 	dbMap *gorp.DbMap
+
+	cac core.CertificateAuthority
+	pub core.Publisher
+
+	clk clock.Clock
+
+	ocspMinTimeToExpiry time.Duration
+	oldestIssuedSCT     time.Duration
+
+	newCertWindow    time.Duration
+	oldOCSPWindow    time.Duration
+	missingSCTWindow time.Duration
+	newCertBatch     int
+	oldOCSPBatch     int
+	missingSCTBatch  int
+
+	numLogs int
+}
+
+// This is somewhat gross but can be pared down a bit once the publisher and this
+// are fully smooshed together
+func newUpdater(log *blog.AuditLogger, stats statsd.Statter, clk clock.Clock,
+	dbMap *gorp.DbMap, ca core.CertificateAuthority, pub core.Publisher,
+	numLogs int, config cmd.OCSPUpdaterConfig) (*OCSPUpdater, error) {
+	updater := OCSPUpdater{
+		stats:   stats,
+		log:     log,
+		clk:     clk,
+		dbMap:   dbMap,
+		cac:     ca,
+		pub:     pub,
+		numLogs: numLogs,
+		// Stuff we initialize
+		rpcMu: new(sync.RWMutex),
+	}
+	if config.NewCertificateBatchSize == 0 || config.OldOCSPBatchSize == 0 || config.MissingSCTBatchSize == 0 {
+		return nil, fmt.Errorf("Batch sizes must be non-zero")
+	}
+	var err error
+	updater.newCertWindow, err = time.ParseDuration(config.NewCertificateWindow)
```

Comment:
NewCertificateWindow and the others like it here can be `cmd.JSONDuration`s now.